src/modalities/models/model.py

```diff
@@ -27,6 +42,42 @@ def forward(self, inputs: Dict[str, torch.Tensor]) -> Dict[str, torch.Tensor]:
     def get_parameters(self) -> Dict[str, torch.Tensor]:
         return {name: param for name, param in self.named_parameters()}
 
+    def _init_weights(self, module: nn.Module, weight_init: WeightInitializationConfig):
+        if isinstance(module, nn.Linear):
+            torch.nn.init.normal_(module.weight, mean=weight_init.mean, std=weight_init.std)
+            if module.bias is not None:
+                torch.nn.init.zeros_(module.bias)
+        elif isinstance(module, nn.Embedding):
+            torch.nn.init.normal_(module.weight, mean=weight_init.mean, std=weight_init.std)
+
+    def initialize_weights(
+        self, weight_init: WeightInitializationConfig, number_of_layers: int, hidden_dim: Optional[int] = None
+    ):
+        # auto: choose std automatically
+        if weight_init.std == "auto":
+            assert hidden_dim is not None, "ERROR! weight_init.std = auto not implemented"
+            weight_init.std = math.sqrt(2 / (5 * hidden_dim))
+
+        # initialize weights
+        self.apply(partial(self._init_weights, weight_init=weight_init))
+
+        if weight_init.type == "plain":
+            pass  # nothing more to do
+        elif weight_init.type in ["scaled", "scaled_embed"]:
+            # apply special scaled init to the residual projections, per GPT-2 paper
+            for pn, p in self.named_parameters():
+                if pn.endswith("c_proj.weight"):
+                    torch.nn.init.normal_(
+                        p, mean=weight_init.mean, std=weight_init.std / math.sqrt(2 * number_of_layers)
+                    )
+            if weight_init.type == "scaled_embed":
+                # apply special init to embeddings, see https://arxiv.org/abs/2312.16903
+                for pn, p in self.named_parameters():
+                    if pn.endswith("wte.weight") or pn.endswith("wpe.weight"):
+                        torch.nn.init.normal_(p, mean=weight_init.mean, std=0.4)
```

Comment:
Where is 0.4 specified in the paper? 

I could only find this reference in the paper: 
"The Scaled Embed is scaling embeddings with an appropriate value. For example, we multiply embeddings by sqrt(d), which is introduced in the original Transformer paper (Vaswani et al., 2017), and then the standard deviations of embeddings become sqrt(2/5)" -> which is equivalent to 0.63.

Reply:
Good catch! I mixed up standard deviation and variance, so `0.4` needs to be replaced by e.g. `math.sqrt(0.4)`, not just here but also for instance in the unit test "test_initialization.py". Will be done as part of #168?